docs/zkapps/tutorials/03-deploying-to-a-network.mdx

```diff
@@ -1,32 +1,456 @@
+
+:::
+
+Now, to finish setting up our code, let's add to our `main.ts`:
+```ts
+...
+  7 } from 'snarkyjs';
+  8
+  9 import fs from 'fs';
+ 10
+ 11 (async function main() {
+...
+ 23   let transactionFee = 100_000_000;
+ 24
+ 25   const configName = process.argv[2];
+ 26
+ 27   const deployerKeysFileContents = fs.readFileSync(
+ 28     'keys/' + configName + '.json',
+ 29     'utf8'
+ 30   );
+ 31
+ 32   const deployerPrivateKeyBase58 = JSON.parse(
+ 33     deployerKeysFileContents
+ 34   ).privateKey;
+ 35
+ 36   const deployerPrivateKey = PrivateKey.fromBase58(deployerPrivateKeyBase58);
+ 37
+ 38   const zkAppPrivateKey = deployerPrivateKey;
+ 39
+ 40   // ----------------------------------------------------
+...
+```
+
+This uses the key generated by the `zk config` tool, stored in `keys/`. The name of the key file will be provided through an argument on the command line (`process.argv[2]`). 
+
+You can run this now with:
+
+```sh
+npm run build && node build/src/main.js berkeley
+```
+
+Which will read keys from `keys/berkeley.json`. Public and Private Keys in Mina are commonly stored in `Base58` for easily readability. In Mina, public keys start with `B62`, and private keys start with `EKE` for easy differentiability.
+
+Our Smart Contract is also stored on chain in the same account we deployed from. So we set `zkAppPrivateKey = deployerPrivateKey`.
+
+### Waiting for accounts to be ready
+
+Next, we will wait for the deployer account to be ready.
+
+In main, we will import a and use the `loopUntilAccountExists` function from `utils.ts`:
+
+```ts
+...
+  9 import fs from 'fs';
+ 10 import { loopUntilAccountExists } from './utils.js';
+ 11
+ 12 (async function main()
+ ...
+  42   // ----------------------------------------------------
+ 43
+ 44   let account = await loopUntilAccountExists({
+ 45     account: deployerPrivateKey.toPublicKey(),
+ 46     eachTimeNotExist: () => {
+ 47       console.log(
+ 48         'Deployer account does not exist. ' +
+ 49           'Request funds at faucet ' +
+ 50           'https://faucet.minaprotocol.com/?address=' +
+ 51           deployerPrivateKey.toPublicKey().toBase58()
+ 52       );
+ 53     },
+ 54     isZkAppAccount: false,
+ 55   });
+ 56
+ 57   console.log(
+ 58     `Using fee payer account with nonce ${account.nonce}, balance ${account.balance}`
+ 59   );
+ 60
+ 61   // ----------------------------------------------------
+ ...
+```
+
+ We wait until the deployer's account exist - if it does not, we share a link with the user to go to the faucet and request funds.
+
+If you've already run `zk deploy` as described earlier in this tutorial, then this should find the account instantly and move on to the next step. But its good practice to have as a check, and useful if you're writing code for other flows, say a CLI application that interacts with an existing zkapp, where the user's key may not have funds yet.
```

Comment:
Given this is only useful when scripting deployment, maybe this block of code should only exist in the utils related to deployment? Opportunity to simplify. Not a strong opinion